src/HDF5MultiDim.chpl

```diff
@@ -277,19 +274,14 @@ module HDF5MultiDim {
             return new MsgTuple(errorMsg, MsgType.ERROR);
         }
 
-        var shape_sym: borrowed GenSymEntry = getGenericTypedArrayEntry(shape_name, st);
+        var shape_sym: borrowed GenSymEntry = getGenericTypedArrayEntry(msgArgs.getValueOf("shape"), st);
         var shape = toSymEntry(shape_sym, int);
 
-        var mode: int;
-        try {
-            mode = mode_str: int;
-        } catch {
-            var errorMsg = "Could not convert mode to numeric";
-            h5tLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-            return new MsgTuple(errorMsg, MsgType.ERROR);
-        }
+        var mode_str = msgArgs.get("mode");
+        var mode: int = mode_str.getIntValue();
```

Comment:
can we make this one line like above

Reply:
Meant to do that after the other compiled. THanks. Updated